pystac/extensions/file.py

```diff
@@ -188,10 +188,11 @@ def data_type(self) -> Optional[List[FileDataType]]:
         Returns:
             FileDataType
         """
-        return map_opt(
-            lambda x: [FileDataType(t) for t in x],
```

Comment:
Is it the case that mypy doesn't allow lambdas? Is there a flag for this? I find this style of short lambda useful and more readable

Reply:
No, it allows lambdas; it's just simpler to type a function. I'm not sure you can type lambdas, but I expect it'd be a mess.

Reply:
Ah right, I think Pyright is OK with inferring types in lambdas without annotations where mypy isn't...a bit of inconvenience but worth it.

Reply:
I think this may actually be an indicator that we're not adequately typing something in the second argument to `map_opt`. I left a comment on `pystac.extensions.eo` to illustrate this in more detail, but I think we may want to keep the untyped lambdas here and track down the typing issues in more detail.

I think the solution from the other comment will also solve this case and allow us to keep the untyped lambda here.